datafusion/optimizer/src/type_coercion.rs

```diff
@@ -426,95 +426,128 @@ impl ExprRewriter for TypeCoercionRewriter {
     }
 }
 
-/// Casts the ScalarValue `value` to coerced type.
-// When coerced type is `Interval` we use `parse_interval` since `try_from_string` not
```

Comment:
The issue was not introduced by this PR, but it seems like it would be a good idea to update `try_from_string()`...